alchemist/alchemist-implementationbase/src/main/kotlin/it/unibo/alchemist/model/implementations/geometry/IntersectionUtils.kt

```diff
@@ -33,80 +25,75 @@ enum class LinesIntersectionType {
  * @param point
  *              the intersection point (if present).
  */
-data class LinesIntersectionResult(
-    /**
-     */
+data class LinesIntersection<P : Vector2D<P>>(
     val type: LinesIntersectionType,
-    /**
-     */
-    val point: Optional<Euclidean2DPosition> = Optional.empty()
-)
+    val point: Optional<P> = Optional.empty()
+) {
+    constructor(point: P) : this(LinesIntersectionType.POINT, Optional.of(point))
+
+    companion object {
+        /**
+         * Creates an instance of [LinesIntersection] whose type is [LinesIntersectionType.EMPTY].
+         */
+        fun <P : Vector2D<P>> empty() = LinesIntersection<P>(LinesIntersectionType.EMPTY)
```

Comment:
why fun and not val?

Reply:
I tried to use val but I need the type parameter, this is incompatible with val to the best of my knowledge.

Reply:
The best of your knowledge is correct.